frontend/src/components/SelectApplication.vue

```diff
@@ -68,10 +71,14 @@ const filteredOptions = computed(() => {
                     <select
                         id="applicationSelect"
                         class="form-select"
-                        v-model="selectedApplication"
+                        :value="JSON.stringify(selectedApplication)"
+                        @change="(e: Event) => setSelectedApplication(e.target ? (e.target as HTMLTextAreaElement).value : null)"
```

Comment:
Looks good to me but not sure if you want to move the whole logic into <script> section?

Reply:
yeah, think it should be fine, moved into the script section